libcore/src/Simulation.cpp

```diff
@@ -187,15 +188,19 @@ bool Simulation::InitArgs()
         return false;
     }
 
-    _em = new EventManager(_config, _building.get(), _config->GetSeed());
-    if(!_em->ReadEventsXml()) {
-        LOG_WARNING("Could not initialize events handling");
+    _em = new EventManager(_building.get());
```

Comment:
Does it need to be a pointer?

Reply:
Changed it to std::unique_ptr<EventManager>